kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -188,25 +189,145 @@ public Transaction build(Engine engine) {
       if (operation == Operation.CREATE_TABLE) {
         throw new TableAlreadyExistsException(table.getPath(engine), "Operation = CREATE_TABLE");
       }
+      return buildTransactionInternal(engine, false /* isNewTableDef */, Optional.of(snapshot));
     } catch (TableNotFoundException tblf) {
       String tablePath = table.getPath(engine);
       logger.info("Table {} doesn't exist yet. Trying to create a new table.", tablePath);
       schema.orElseThrow(() -> requiresSchemaForNewTable(tablePath));
-      // Table doesn't exist yet. Create an initial snapshot with the new schema.
-      Metadata metadata = getInitialMetadata();
-      Protocol protocol = getInitialProtocol();
-      SnapshotQueryContext snapshotContext = SnapshotQueryContext.forVersionSnapshot(tablePath, -1);
-      LogReplay logReplay =
-          getEmptyLogReplay(engine, metadata, protocol, snapshotContext.getSnapshotMetrics());
-      snapshot =
-          new InitialSnapshot(table.getDataPath(), logReplay, metadata, protocol, snapshotContext);
+      return buildTransactionInternal(engine, true /* isNewTableDef */, Optional.empty());
     }
+  }
+
+  /**
+   * Returns a built {@link Transaction} for this transaction builder (with the input provided by
+   * the user) given the provided parameters. This includes validation and updates as defined in the
+   * builder.
+   *
+   * @param isNewTableDef whether we are defining a new table definition or not. This determines
+   *     what metadata to commit in the returned transaction, and what operations to allow or block.
+   * @param existingSnapshot the existing latest snapshot of the table (or empty for a new table).
```

Comment:
"empty for a new table" what will it be for replace table case? `isNewTableDef` is true and existingSnapshot is present?

Reply:
updating the docs